applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAXNativeAdView.java

```diff
@@ -103,42 +103,42 @@ public AppLovinMAXNativeAdView(final int viewId,
 
             if ( "addTitleView".equals( call.method ) )
             {
-                addTitleView( call );
+                if ( nativeAd != null ) addTitleView( call, nativeAd );
```

Comment:
I think we'd still have the same problem as before. We should skip the null check here, and pass it into the method and do the `null` check there instead. 

Same for other methods below.

Reply:
@alhiwatan - The issue is this code is not thread safe and `nativeAd` needs to be used in an atomic manner.

By extracting it into a local variable and having your code reference that local varaible, you prevent a potential race condition such that if `if ( nativeAd != null )` is `true`, but right after `nativeAd` is set to `null`. That's the issue of referencing the class field instead of a local var.

Reply:
@thomasmso, thanks, please review the update.